source/common/http/http1/codec_impl.cc

```diff
@@ -436,12 +434,16 @@ void ConnectionImpl::completeLastHeader() {
     sendProtocolError(Http1ResponseCodeDetails::get().TooManyHeaders);
     const absl::string_view header_type =
         processing_trailers_ ? Http1HeaderTypes::get().Trailers : Http1HeaderTypes::get().Headers;
-    throw CodecProtocolException(absl::StrCat(header_type, " size exceeds limit"));
+    codec_exception_ = absl::InternalError(absl::StrCat(header_type, " size exceeds limit"));
```

Comment:
Should completeLastHeader return Status instead?
Call sites would need to save the status so it can get across the parser.

Reply:
Yes, I agree -- we won't always have matching HPE_ errors for the statuses like we did in this case. However, making this change means I'll need StatusOr (it is long-winded, but `onHeadersCompleteBase()` would also need absl::Status return value, but this also has a special return value when it needs to stop accepting further data.